<?php

namespace Wellid\Exception;

/**
 * Generated by PHPUnit_SkeletonGenerator on 2016-02-07 at 11:36:24.
 */
class FileNotFoundTest extends \PHPUnit_Framework_TestCase {
    /**
     * Sets up the fixture, for example, opens a network connection.
     * This method is called before a test is executed.
     */
    protected function setUp() {
    }

    /**
     * Tears down the fixture, for example, closes a network connection.
     * This method is called after a test is executed.
     */
    protected function tearDown() {
        
    }

    /**
     * @covers Wellid\Exception\FileNotFound::__construct
     */
    public function testFileNotFound() {
        $e = new FileNotFound('main.css', 'Style Sheet', 500);
        $this->assertInstanceOf('\Exception', $e);
        $this->assertEquals(__LINE__-2, $e->getLine());
        $this->assertEquals(__FILE__, $e->getFile());
        $this->assertEquals(500, $e->getCode());
        $this->assertNull($e->getPrevious());
        try {
            throw $e;
        } catch (Exception $ex) {
            $this->assertContains('main.css', $ex->getMessage());
            $this->assertContains('Style Sheet', $ex->getMessage());
            return;
        }
        $this->fail('Exception was not thrown');
    }    
}
